src/screens/Exchange/Exchange.js

```diff
@@ -972,13 +975,15 @@ class ExchangeScreen extends React.Component<Props, State> {
 
   generateSupportedAssetsOptions = (assets: Asset[]) => {
     if (!Array.isArray(assets)) return [];
-    const { balances, baseFiatCurrency } = this.props;
+    const { balances, baseFiatCurrency, rates } = this.props;
     return [...assets] // prevent mutation of param
       .map(({ symbol, iconUrl, ...rest }) => {
         const rawAssetBalance = getBalance(balances, symbol);
         const assetBalance = rawAssetBalance ? formatAmount(rawAssetBalance) : null;
-        const assetBalanceInFiat = rawAssetBalance ? formatAmount(getBalance(balances, symbol)) : null;
-        const formattedBalanceInFiat = assetBalanceInFiat ? formatFiat(assetBalanceInFiat, baseFiatCurrency) : null;
+        const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+        const assetBalanceInFiat = assetBalance ?
+          parseFloat(assetBalance) * getRate(rates, symbol, fiatCurrency) : null;
+        const formattedBalanceInFiat = assetBalanceInFiat ? formatFiat(assetBalanceInFiat, fiatCurrency) : null;
```

Comment:
maybe add a function to avoid duplicating code?

Reply:
added in 797828a